### ISSUE TITLE
[java] ImplicitFunctionalInterface reports false positive for interfaces with inherited abstract methods

### ISSUE THREAD

**Issue by cowwoc** [pmd/pmd#6111]
**Affects PMD Version:** 7.17.0

**Rule:** ImplicitFunctionalInterface

Please provide the rule name and a link to the rule documentation:
<https://docs.pmd-code.org/pmd-doc-7.17.0/pmd_rules_java_bestpractices.html#implicitfunctionalinterface>

**Description:**

The `ImplicitFunctionalInterface` rule incorrectly reports a violation for interfaces that inherit multiple abstract methods from parent interfaces, suggesting they should be annotated with `@FunctionalInterface`. A functional interface must have
 exactly one abstract method per JLS §9.8, but interfaces that inherit multiple abstract methods cannot be functional interfaces and should not trigger this rule.

**Code Sample demonstrating the issue:**

```java
// Parent Interface (FormattingRule.java)
package io.github.cowwoc.styler.formatter.api;

import java.time.Duration;

public interface FormattingRule {
    String getRuleId();
    int getPriority();
    RuleConfiguration getDefaultConfiguration();
    ValidationResult validate(FormattingContext context);
    FormattingResult apply(FormattingContext context);

    default Duration getMaxExecutionTime() {
        return Duration.ofSeconds(5);
    }

    default long getMaxMemoryUsage() {
        return 100 * 1024 * 1024;
    }
}

// Child Interface (MutableFormattingRule.java)
package io.github.cowwoc.styler.formatter.api.plugin;

import io.github.cowwoc.styler.formatter.api.FormattingRule;
import io.github.cowwoc.styler.formatter.api.MutableFormattingContext;

public interface MutableFormattingRule extends FormattingRule {
    void format(MutableFormattingContext context);
}
```
**Expected outcome**:

PMD should not report a violation for `MutableFormattingRule` at line 12. The interface has 6 abstract methods total (5 inherited from FormattingRule + 1 declared: format()), which means it cannot be annotated with `@FunctionalInterface`. The rule
should only suggest `@FunctionalInterface` for interfaces with exactly one abstract method. This is a false positive because attempting to add `@FunctionalInterface` would result in a compilation error.

**Running PMD through**: Maven (maven-pmd-plugin 3.26.0)

**Comment by oowekyala** [comment]
This is probably a case where the class file for FormattingRule is not on the auxclasspath, because the rule definitely handles interfaces with inherited abstract methods properly if all types are available. But we could make the behavior of the rule err towards false negatives rather than false positives when some supeclasses are unresolved.

**Comment by cowwoc** [comment]
That's odd. I am using the Maven plugin and the classes in question are both part of the project. Doesn't the Maven plugin place the local project's files on the auxclasspath? If it can see one source-code file then it should definitely be able to see the other...

**Comment by adangel** [comment]
When is PMD executed? Does maven's compile phase run before pmd is executed? It's important, that the source is compiled before running PMD...

**Comment by cowwoc** [comment]
You were right. PMD was bound to the `validate` phase instead of `verify`. Changing this resolved the problem.
